lib/ansible/module_utils/basic.py

```diff
@@ -2327,6 +2327,8 @@ def _return_formatted(self, kwargs):
                 for d in kwargs['deprecations']:
                     if isinstance(d, SEQUENCETYPE) and len(d) == 2:
                         self.deprecate(d[0], version=d[1])
+                    elif isinstance(d, Mapping):
+                        self.deprecate(d['msg'], version=d.get('version', None))
```

Comment:
what if msg does not exist?, shouldn't it also be a get

Reply:
in any caze, deprecations should always be a list, unsure how a dict is making its way here

Reply:
Both python and powershell module_utils define it as a list of dictionaries.

Reply:
ah, nvmd, misread the code, thought you were operating on the 'top level' ... yes it is a list of dictionaries but we allowed for pre-existing string cases

Reply:
unsure why we need mapping, need to look at original bug, the self.deprecate(d) should be able to handle dicts already

Reply:
So I guess this should be backported as well.

Reply:
If the `module.deprecate()` call was used, the message cannot be missing. And we rewrote all modules to use this call.

Reply:
yeah, it makes sense now, i was just being dumb